docs/design/pattern_matching.md

```diff
@@ -257,9 +226,9 @@ fn F(T:! type, x: T) {
 }
 ```
 
-The `template` keyword indicates the binding pattern is introducing a template
-binding, so name lookups into the binding will not be fully resolved until its
-value is known.
+This form behaves the same as a runtime binding pattern, except that the
+scrutinee is expected to have symbolic phase, or template phase if the
+`template` form is used. This form cannot appear inside a `var` pattern.
```

Comment:
I don't believe this is true. The result of the binding has either symbolic or template phase depending whether `template` appears on the binding, but not the scrutinee. See #2153.

Reply:
OK, then we agree on the intent, and the problem is that what I wrote isn't clear. The "scrutinee is expected" phrasing is intended to hook into the phrasing of the section on runtime binding patterns, which says "The scrutinee is expected to be a runtime expression..." and later says that the expression is converted to satisfy that expectation (among others).

In other words, I'm trying to say something like "Same as the previous section, but replace 'runtime' with 'symbolic'/'template'." Maybe the best option is to combine the two sections, and break down the expected phase into cases, just like the expected category already is?

Reply:
I find that exposition unclear. If you say "the scrutinee is expected to have type `i32`" then I read that as a diagnostic will be issued if the scrutinee expression has a different type. To me, the scrutinee is an expression the user wrote in the code.

Reply:
OK, how about this?